### PR TITLE
Add type annotations for tmt/cli.py

### PR DESCRIPTION
**Pull request by thrix**
Signed-off-by: Miroslav Vadkerti <mvadkert@redhat.com>

### CODE REVIEW

**Comment by thrix** [comment]
Not ready for review yet, we want to introduce a container for the click context object

File: tmt/cli.py [hunk 1]
```diff
@@ -338,9 +366,10 @@ def run_tests(context, **kwargs):
     callback = run.resultcallback
 
 
-@callback()
+# TODO: commands is unknown, needs revisit
+@callback()  # type: ignore
```
Comment (thrix) [inline tmt/cli.py:370]: @happz any idea about this one? I am unable to convince mypy about the type of this decorator:

```
tmt/cli.py:369: error: Untyped decorator makes function "finito" untyped
```

Comment (happz) [reply tmt/cli.py:370]: Probably not much we can do about it, we'd have to annotate the decorator... I'd add `ignore` and move on.

Comment (thrix) [reply tmt/cli.py:370]: thanks

File: tmt/cli.py [hunk 2]
```diff
@@ -53,31 +74,32 @@ def get_command(self, context, cmd_name):
...
         context.fail('Did you mean {}?'.format(
             listed(sorted(matches), join='or')))
+        return None
 
 
 # ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 #  Common Options
 # ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-def verbose_debug_quiet(function):
```
Comment (janhavlin) [inline tmt/cli.py:62]: Could these options functions be improved to `def verbose_debug_quiet(function: Callable[..., Any]) -> Callable[..., Any]`?

Comment (thrix) [reply tmt/cli.py:62]: thanks, will check

Comment (thrix) [reply tmt/cli.py:62]: changed

File: tmt/convert.py [hunk 3]
```diff
@@ -67,10 +67,10 @@ def read_manual(
 
     try:
         if plan_id:
-            all_cases = nitrate.TestPlan(int(plan_id)).testcases
+            all_cases = nitrate.TestPlan(plan_id).testcases
```
Comment (lukaszachy) [inline tmt/convert.py:70]: Please add type to the click options. Otherwise runtime propagates 'str' here which is not accepted.

Comment (thrix) [reply tmt/convert.py:70]: +1, I was just looking into why it does not work :) thanks

Comment (thrix) [reply tmt/convert.py:70]: added type to the parameters

Comment (thrix) [reply tmt/convert.py:70]: waiting for tests

Comment (thrix) [reply tmt/convert.py:70]: all good \o/

**Review by lukaszachy — APPROVED** [review]
LGTM

**Review by janhavlin — APPROVED** [review]

**Pull request merged**